components/ai_chat/content/browser/ai_chat_tab_helper.h

```diff
@@ -6,179 +6,41 @@
-                      print_preview_extraction_delegate);
-
-  // content::WebContentsObserver
-  void NavigationEntryCommitted(
-      const content::LoadCommittedDetails& load_details) override;
-  void TitleWasSet(content::NavigationEntry* entry) override;
-  void DidFinishLoad(content::RenderFrameHost* render_frame_host,
-                     const GURL& validated_url) override;
-
-  // ai_chat::AssociatedContentDriver
-  void GetPageContent(FetchPageContentCallback callback,
-                      std::string_view invalidation_token) override;
-  void OnNewPage(int64_t navigation_id) override;
-
-  // Called when an event of significance occurs that, if the page is a
-  // same-document navigation, should result in that previous navigation
-  // being considered as a new page.
-  void MaybeSameDocumentIsNewPage();
-
-  void GetSearchSummarizerKey(GetSearchSummarizerKeyCallback callback) override;
-
-  bool HasOpenAIChatPermission() const override;
-
-  void GetScreenshots(
-      mojom::ConversationHandler::GetScreenshotsCallback callback) override;
-
-  void OnScreenshotsCaptured(
-      mojom::ConversationHandler::GetScreenshotsCallback callback,
-      base::expected<std::vector<std::vector<uint8_t>>, std::string>);
-
-  void OnFetchPageContentComplete(FetchPageContentCallback callback,
-                                  std::string content,
-                                  bool is_video,
-                                  std::string invalidation_token);
-
-#if BUILDFLAG(ENABLE_PDF)
-  void OnPDFDocumentLoadComplete(FetchPageContentCallback callback);
-
-  void OnGetPDFPageCount(FetchPageContentCallback callback,
-                         pdf::mojom::PdfListener::GetPdfBytesStatus status,
-                         const std::vector<uint8_t>& bytes,
-                         uint32_t page_count);
-
-  void OnAllPDFPagesTextReceived(
-      FetchPageContentCallback callback,
-      const std::vector<std::pair<size_t, std::string>>& page_texts);
-#endif  // BUILDFLAG(ENABLE_PDF)
-
-
-  void SetPendingGetContentCallback(FetchPageContentCallback callback);
-
-  raw_ptr<AIChatMetrics> ai_chat_metrics_;
-
-  bool is_same_document_navigation_ = false;
-  int pending_navigation_id_;
-  std::u16string previous_page_title_;
-  bool is_page_loaded_ = false;
-
-  // TODO(petemill): Use signal to allow for multiple callbacks
-  FetchPageContentCallback pending_get_page_content_callback_;
-
-  std::unique_ptr<PrintPreviewExtractionDelegate>
-      print_preview_extraction_delegate_;
-  std::unique_ptr<PageContentFetcherDelegate> page_content_fetcher_delegate_;
+  AIChatTabHelper(
+      content::WebContents* web_contents,
+      std::unique_ptr<
+          AssociatedWebContentsContent::PrintPreviewExtractionDelegate>
+          print_preview_extraction_delegate);
 
-  std::unique_ptr<FullScreenshotter> full_screenshotter_;
+  std::unique_ptr<AssociatedWebContentsContent> web_contents_content_;
```

Comment:
FWIW I think this name now sufficiently disambiguates from WebContents and does not feel like something is being associated with this tab helper now. So, whilst not perfect, I do think it's better and good